seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SeaTunnelRow.java

```diff
@@ -33,6 +34,8 @@ public final class SeaTunnelRow implements Serializable {
     /** The array to store the actual internal format values. */
     private final Object[] fields;
 
+    private Map<String, Object> metadata = new HashMap<String, Object>();
```

Comment:
Please use options in https://github.com/apache/seatunnel/blob/d7798a6dbacba34ea9e80edbc4f8717b974d74a1/seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SeaTunnelRow.java#L39